tests/base/src/test/java/org/wildfly/security/auth/client/MaskedPasswordSaslAuthenticationTest.java

```diff
@@ -0,0 +1,134 @@
+            WildFlyElytronSaslPlainProvider.getInstance(),
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+
+    @BeforeClass
+    public static void registerProvider() {
+        for (Provider provider : providers) {
+            Security.insertProviderAt(provider, 1);
+        }
+    }
+
+    @AfterClass
+    public static void removeProvider() {
+        for (Provider provider : providers) {
+            Security.removeProvider(provider.getName());
+        }
+    }
+
+    /**
+     * Test a successful exchange using the PLAIN mechanism where no Authorization ID is specified and password in client
+     * XML config is specified as a masked password.
+     */
+    @Test
+    public void testSuccessfulExchangeWithXmlConfig() throws Exception {
+        SaslServer server = new SaslServerBuilder(PlainSaslServerFactory.class, PLAIN)
+                                .setUserName(USERNAME)
+                                .setPassword(PASSWORD.toCharArray())
+                                .build();
+
+        //Creating SASL client from XML configuration file
+        System.setProperty("wildfly.config.url", MaskedPasswordSSLAuthenticationTest.class.getResource(CONFIG_FILE).toExternalForm());
+        AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Security.insertProviderAt(WildFlyElytronPasswordProvider.getInstance(), 1));
+        AuthenticationContext context = AuthenticationContext.getContextManager().get();
+
+        AuthenticationContextConfigurationClient contextConfigurationClient = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
+        SaslClient client = contextConfigurationClient.createSaslClient(new URI(CONFIG_FILE), context.authRules.configuration, Arrays.asList(new String[]{PLAIN}));
+
+        assertTrue(client.hasInitialResponse());
+        byte[] message = client.evaluateChallenge(new byte[0]);
+        assertEquals("\0"+USERNAME+"\0"+PASSWORD,new String(message, StandardCharsets.UTF_8));
+
+        server.evaluateResponse(message);
+        assertTrue(server.isComplete());
+        assertTrue(client.isComplete());
+        assertEquals(USERNAME, server.getAuthorizationID());
+    }
+
+    @Test
+    public void testSuccessfulExchangeWithProgrammaticConfig() throws Exception {
+        SaslServer server = new SaslServerBuilder(PlainSaslServerFactory.class, PLAIN)
+                .setUserName(USERNAME)
+                .setPassword(PASSWORD.toCharArray())
+                .build();
+
+        //Creating SASL client from authentication configuration programmatically
+        AuthenticationConfiguration maskedConfig =
+                AuthenticationConfiguration.empty()
+                        .setSaslMechanismSelector(SaslMechanismSelector.NONE.addMechanism(PLAIN))
+                        .useName(USERNAME)
+                        .usePassword("YFBlotObdCo=", null, null, "100", "12345678", null);
```

Comment:
It would be good to have tests for the other similar methods that were added.

Reply:
this method calls the other two methods in its implementation